File: src/main/java/bad/robot/radiate/ui/FrameFactory.java
package bad.robot.radiate.ui;

import java.awt.*;
import java.util.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.stream.Stream;

import static java.awt.GraphicsEnvironment.getLocalGraphicsEnvironment;
import static java.util.stream.Collectors.toList;
import static java.util.stream.IntStream.range;

@Deprecated
public interface FrameFactory {

    Collection<StatusFrame> create();

    static FrameFactory fullScreen() {
        return () -> {
            GraphicsDevice[] screens = getLocalGraphicsEnvironment().getScreenDevices();
            Stream<StatusFrame> frames = range(0, screens.length).mapToObj(index -> new StatusFrame(index, new FullScreen(screens[index].getDefaultConfiguration().getBounds())));
            return frames.collect(toList());
        };
    }

    static FrameFactory desktopMode() {
        return () -> {
            List<StatusFrame> frames = new ArrayList<>(1);
            Rectangle bounds = getLocalGraphicsEnvironment().getDefaultScreenDevice().getDefaultConfiguration().getBounds();
            frames.add(new StatusFrame(0, new DesktopMode(bounds)));
            return frames;
        };
    }
}

File: src/main/scala/bad/robot/radiate/ui/FrameFactory.scala
package bad.robot.radiate.ui

import java.awt.GraphicsEnvironment.getLocalGraphicsEnvironment

trait FrameFactory {
  def create: List[StatusFrame]
}

object FrameFactory {

  def fullScreen = new FrameFactory {
    def create = {
      val screens = getLocalGraphicsEnvironment.getScreenDevices
      val frames = (0 until screens.length).map(index => new StatusFrame(index, new FullScreen(screens(index).getDefaultConfiguration.getBounds)))
      frames.toList
    }
  }

  def desktopMode = new FrameFactory {
    def create = {
      val bounds = getLocalGraphicsEnvironment.getDefaultScreenDevice.getDefaultConfiguration.getBounds
      val frames = Array(new StatusFrame(0, new DesktopMode(bounds)))
      frames.toList
    }
  }
}
